python/ray/train/jax/jax_trainer.py

```
@@ -0,0 +1,205 @@
+            by the ``preprocessor`` if one is provided.
+        preprocessor: A ``ray.data.Preprocessor`` to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        train_loop_per_worker: Union[Callable[[], None], Callable[[Dict], None]],
+        *,
+        train_loop_config: Optional[Dict] = None,
+        jax_config: Optional[JaxConfig] = None,
+        scaling_config: Optional[ScalingConfig] = None,
+        dataset_config: Optional[Dict[str, DatasetConfig]] = None,
+        run_config: Optional[RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
+        preprocessor: Optional["Preprocessor"] = None,
+        resume_from_checkpoint: Optional[Checkpoint] = None,
+    ):
+        if not jax_config:
+            jax_config = JaxConfig()
+
+        super(JaxTrainer, self).__init__(
+            train_loop_per_worker=train_loop_per_worker,
+            train_loop_config=train_loop_config,
+            backend_config=jax_config,
+            scaling_config=scaling_config,
+            dataset_config=dataset_config,
+            run_config=run_config,
+            datasets=datasets,
+            preprocessor=preprocessor,
+            resume_from_checkpoint=resume_from_checkpoint,
+        )
+
+    @classmethod
+    def _validate_scaling_config(cls, scaling_config: ScalingConfig) -> ScalingConfig:
+        """Return scaling config dataclass after validating updated keys."""
+        ensure_only_allowed_dataclass_keys_updated(
+            dataclass=scaling_config,
+            allowed_keys=cls._scaling_config_allowed_keys,
+        )
+
+        use_gpu = scaling_config.use_gpu
+        use_tpu = (
+            scaling_config.resources_per_worker
+            and scaling_config.resources_per_worker.get("TPU", 0) > 0
+        )
+
+        # cpu parallelism is not supported in jax
+        if not (use_gpu or use_tpu):
```

Can we raise an error if both `use_tpu` and `use_gpu` are True?